extension/encoding/awscloudwatchmetricstreamsencodingextension/json_unmarshaler.go

```diff
@@ -139,32 +138,17 @@ func (c *formatJSONUnmarshaler) UnmarshalMetrics(record []byte) (pmetric.Metrics
 	for datumIndex := 0; scanner.Scan(); datumIndex++ {
 		var cwMetric cloudwatchMetric
 		if err := jsoniter.ConfigFastest.Unmarshal(scanner.Bytes(), &cwMetric); err != nil {
-			c.logger.Error(
-				"Unable to unmarshal input",
-				zap.Error(err),
-				zap.Int("datum_index", datumIndex),
-			)
-			continue
+			return pmetric.Metrics{}, fmt.Errorf("error unmarshaling datum at index %d: %w", datumIndex, err)
 		}
 		if err := validateMetric(cwMetric); err != nil {
-			c.logger.Error(
-				"Invalid metric",
-				zap.Int("datum_index", datumIndex),
-				zap.Error(err),
-			)
-			continue
+			return pmetric.Metrics{}, fmt.Errorf("invalid cloudwatch metric at index %d: %w", datumIndex, err)
```

Comment:
QQ: I didn't verify it on receiver side where we will use this encoding extension, Do we log these errors already on the awsfirehosereceiver for operational visibility in addition to propagating these errors back to clients ?

Reply:
If we do, it should be at debug level, otherwise a faulty client can cause non-actionable log spam in a perfectly well configured collector.

Reply:
Agreed. Just want to make sure we have the ability to debug issues on the collector side.

Reply:
this is also a change in the sense that we used to return the partially successful unmarshalled metrics. Now we fail hard on first error.

Reply:
Yes, this is a new component though - rationale for the change is in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/38596